boards/arm/stm32f030_demo/doc/stm32f030_demo.rst

```diff
@@ -0,0 +1,26 @@
+.. _stm32f030_demo:
+
+STM32F030 DEMO BOARD
+####################
+
+This board has the bare minimum components required to power on
+the STM32F030F4P6 MCU.
+For practical use, you'll need to add additional components
+and circuits using a breadboard, for example.
+
+.. image:: img/stm32f030_demo.jpg
+   :width: 345px
+   :align: center
+   :height: 592px
+   :alt: STM32F030 DEMO BOARD
+
+More information about the board can be found at the `stm32-base.org website`_.
+
```

Comment:
It would be good to have a least information about how to flash

Reply:
Can you gleen something from https://docs.zephyrproject.org/latest/boards/arm/stm32f0_disco/doc/index.html#programming-and-debugging to add here? Or does "bare minimum components required to power on" not include flashing?  Can this board run blinky as an example?

Reply:
@erwango, @dbkinder

> It would be good to have a least information about how to flash
> Can you gleen something from https://docs.zephyrproject.org/latest/boards/arm/stm32f0_disco/doc/index.html#programming-and-debugging to add here?
> Or does "bare minimum components required to power on" not include flashing?

Discovery boards have embedded st-link programmer. The STM32F030 DEMO board does not include an embedded programmer. The OLIMEX-STM32-E407 board has no embedded programmer either. I'll take a look on the OLIMEX-STM32-E407 board documentation and add adapt it for STM32F030 DEMO.

> Can this board run blinky as an example?

Please see the https://github.com/zephyrproject-rtos/zephyr/pull/15560/commits/4121fd1172265ab83273b6a31abd6acc5d5fcd47 commit message. This board runs ``basic/blinky`` example.
I suppose I have to add example list  to the pull request message. Alas example list is very short. E.g. the 4K RAM makes it impossible to run ```philosophers``` example.

Reply:
@dbkinder 

I have updated documentation flashing chapter.
Please review.